File: app/src/main/java/com/binglivewallpaper/BootReceiver.kt
package com.binglivewallpaper

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

/**
 * Re-schedules the daily refresh work after a device reboot.
 */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            Log.d(TAG, "BOOT_COMPLETED received; scheduling refresh work")
            BingRefreshWorker.schedule(context)
        }
    }

    companion object {
        private const val TAG = "BootReceiver"
    }
}

File: app/src/main/clojure/com/binglivewallpaper/boot_receiver.clj
(ns com.binglivewallpaper.boot-receiver
  (:require [com.binglivewallpaper.refresh-worker :as refresh-worker])
  (:import [android.content Context Intent]
           [android.util Log])
  (:gen-class
   :name com.binglivewallpaper.BootReceiver
   :extends android.content.BroadcastReceiver
   :prefix "receiver-"))

(def ^:private tag "BootReceiver")

(defn receiver-onReceive
  [_ ^Context context ^Intent intent]
  (when (= (.getAction intent) Intent/ACTION_BOOT_COMPLETED)
    (Log/d tag "BOOT_COMPLETED received; scheduling refresh work")
    (refresh-worker/schedule-work context)))
